src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -615,16 +623,48 @@ public void StateChanged(long index)
             }
         }
 
-        private void InitializeFromDatabaseRecord(DatabaseRecord dbRecord)
+        private void NotifyFeaturesAboutStateChange(DatabaseRecord record, long index)
         {
-            if (dbRecord == null)
-                return;
+            InitializeFromDatabaseRecord(record);
+
+            TransformerStore.HandleDatabaseRecordChange(record);
+            IndexStore.HandleDatabaseRecordChange(record, index);
+            ReplicationLoader?.HandleDatabaseRecordChange(record);
+            EtlLoader?.HandleDatabaseRecordChange(record);
+            OnDatabaseRecordChanged(record);
+        }
 
+        private void NotifyFeaturesAboutValueChange(DatabaseRecord record, long index)
+        {
+            SubscriptionStorage?.HandleDatabaseValueChange(record);
+        }
+
+        public void RefreshFeatures()
+        {
+            if (_databaseShutdown.IsCancellationRequested)
+                ThrowDatabaseShutdown();
+
+            DatabaseRecord record = null;
+            long index;
+            using (_serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                record = _serverStore.Cluster.ReadDatabase(context, Name,out index);
+            }
+            NotifyFeaturesAboutStateChange(record, index);
+            NotifyFeaturesAboutValueChange(record, index);
+        }
+
+        private void InitializeFromDatabaseRecord(DatabaseRecord dbRecord)
+        {
             lock (this)
```

Comment:
This lock isn't enough. The older version might run later, resulting in us seeing stale data.